client/src/components/common/challenge/index.tsx

```diff
@@ -6,17 +6,23 @@ import styles from './challenge.module.scss';
 import { IChallenge } from './types';
 import { CollectionModal } from 'components/modals';
 
-const Challenge: React.FC<IChallenge> = ({ linkToAuthor, author, stats, title, rank, tags }) => {
+const Challenge: React.FC<
+	IChallenge & {
+		showAddToCollection?: boolean;
+	}
+> = ({ linkToAuthor, author, stats, title, rank, tags, showAddToCollection }) => {
 	const [isOpen, setIsOpen] = React.useState(false);
 	return (
 		<div className={styles.challenge}>
 			<ChallengeHeader title={title} rank={rank} linkToAuthor={linkToAuthor} />
 			<ChallengeStats stats={{ ...stats, author }} />
 			{tags && Boolean(tags.length) ? <ChallengeTagsList tags={tags} /> : null}
 			{isOpen ? <CollectionModal isOpen={isOpen} setIsOpen={setIsOpen} /> : null}
-			<a className={styles.modal} onClick={() => setIsOpen(true)}>
-				Add to collection
-			</a>
+			{showAddToCollection === undefined || showAddToCollection === true ? (
+				<a className={styles.modal} onClick={() => setIsOpen(true)}>
+					Add to collection
+				</a>
+			) : null}
```

Comment:
You can set showAddToCollection = true as the default value if it's undefined and then just write 
{showAddToCollection && <a className={styles.modal} onClick={() => setIsOpen(true)}>Add to collection</a>}
not necessary but would be better I think, other is okay